src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,229 @@
+package duke;
+
+import duke.exceptions.EmptyDescriptionException;
+import duke.exceptions.EmptyTimeFieldException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.util.Scanner;
+
+/**
+ * Chatbot program Duke that tracks user tasks.
+ *
+ * @author Leow Yuan Yang
+ * @version 4.0
+ * @since 2021-08-25
+ */
+public class Duke {
+    private static final String SEPARATOR = "\t==============================================";
+    private static final String logoArt = "\t                                        \n" +
+            "\t             ..........                 \n" +
+            "\t           .,,..,.,,,,                  \n" +
+            "\t           ,.,,,,.,...                  \n" +
+            "\t          ,.......,*,/& .               \n" +
+            "\t          ,..,..,((/**(((,              \n" +
+            "\t           *(#%(((#/#((/       *        \n" +
+            "\t         /(%#########%##(*,,,((*        \n" +
+            "\t              ((#(((((#      (((*       \n" +
+            "\t        ((((((   /((((/,., .... #       \n" +
+            "\t         (((#    .(((..*...,*///        \n" +
+            "\t         ((/      .,,**  ..**///        \n" +
+            "\t                  .,*,..  ,**//         \n" +
+            "\t                ..,,,..,/ ,*//          \n" +
+            "\t             *,,****,        " ;
+    private static final String nameArt = "\t _______  __   __  __   __  ______    _______ \n" +
+            "\t|       ||  |_|  ||  | |  ||    _ |  |       |\n" +
+            "\t|  _____||       ||  | |  ||   | ||  |    ___|\n" +
+            "\t| |_____ |       ||  |_|  ||   |_||_ |   |___ \n" +
+            "\t|_____  ||       ||       ||    __  ||    ___|\n" +
+            "\t _____| || ||_|| ||       ||   |  | ||   |    \n" +
+            "\t|_______||_|   |_||_______||___|  |_||___|    ";
+    private static Task[] taskList = new Task[100];
+
+    public static void main(String[] args) {
+        boolean isDone = false;
+        Scanner in = new Scanner(System.in);
+
+        printWelcomeMessage();
+
+        do {
+            // Read inputs and split them into substrings
+            String[] userInputs = readUserInput(in);
+            String command = userInputs[0];
+
+            switch (command) {
+            case "bye":
+                isDone = true;
+                printExitMessage();
+                break;
+            case "list":
+                printTaskList(taskList);
+                break;
+            case "done":
+                try {
+                    executeDone(userInputs);
+                } catch (Exception exception) {
+                    printErrorMessage();
```

Comment:
Maybe the printed message can contain more information about the error to have a better user experience? For example, include the exception content?